src/ghdcbot/adapters/github/app_auth.py

```
@@ -0,0 +1,175 @@
+When ``GITHUB_APP_ID``, ``GITHUB_APP_INSTALLATION_ID``, and a private key are set,
+Gitcord uses short-lived installation tokens (``GitcordApp[bot]``) instead of a PAT.
+"""
+
+from __future__ import annotations
+
+import logging
+import os
+import time
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable
```

Ruff (0.16.0)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)